apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableByteString.cs

```diff
@@ -0,0 +1,301 @@
+        /// </param>
+        public BigtableByteString(string valueAsUtf8)
+        {
+            _value = valueAsUtf8 != null ? ByteString.CopyFromUtf8(valueAsUtf8) : ByteString.Empty;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="BigtableByteString"/> value from the big-endian encoding of a signed 64-bit value.
+        /// </summary>
+        /// <param name="value">The 64-bit value which should be used to initialize the byte string.</param>
+        public BigtableByteString(long value)
+        {
+            var bytes = BitConverter.GetBytes(value);
+            if (BitConverter.IsLittleEndian)
+            {
+                Array.Reverse(bytes);
+            }
+            _value = ByteString.CopyFrom(bytes);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="BigtableByteString"/> value from an array of bytes.
+        /// </summary>
+        /// <param name="value">
+        /// The bytes making up the byte string. May be null in which case the empty byte string is used.
+        /// </param>
+        public BigtableByteString(params byte[] value)
+        {
+            _value = value != null ? ByteString.CopyFrom(value) : ByteString.Empty;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="BigtableByteString"/> value from a byte string.
+        /// </summary>
+        /// <param name="value">
+        /// The byte string making up the value. May be null in which case the empty byte string is used.
+        /// </param>
+        public BigtableByteString(ByteString value)
+        {
+            _value = value ?? ByteString.Empty;
+        }
+
+        /// <summary>
+        /// Gets the number of bytes in the byte string.
+        /// </summary>
+        public int Length => Value.Length;
+
+        /// <summary>
+        /// Gets the underlying byte string.
+        /// </summary>
+        public ByteString Value => _value ?? ByteString.Empty;
+
+        /// <summary>
+        /// Returns the byte at index <paramref name="index"/>.
+        /// </summary>
+        /// <param name="index">
+        /// The index in the byte string to return. Must be greater than or equal to 0, and less than <see cref="Length"/>.
+        /// </param>
+        /// <returns>The byte at index <paramref name="index"/>.</returns>
+        public byte this[int index] => Value[index];
+
+        /// <inheritdoc />
+        public override bool Equals(object obj) => obj is BigtableByteString other && Equals(other);
+
+        /// <inheritdoc />
+        public override int GetHashCode() => Value.GetHashCode();
+
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            var value = Value;
+            var builder = new StringBuilder($"{nameof(BigtableByteString)}: Length={value.Length}; ");
+            int initialLength = builder.Length;
+
+            // For short byte strings which can be interpreted as simple strings, display them that way.
```

Comment:
Hmm. I'm not sure whether I'd do this. It would be weird to have a BigtableByteString based on a counter (for example) which kept changing representation. I'd just go for hex all the time, I think.

Reply:
Yeah, that makes sense. I was trying to make the debugging experience nicer for anyone using readable strings for row keys, column names, or cell values, but I guess it kind of hurts the experience in other ways.